crates/storage/src/transactional.rs

```diff
@@ -33,23 +33,29 @@ use crate::{
     kv_store::KVItem,
 };
 
-/// Provides a view of the storage at the given height.
+/// Provides a view of the storage at the latest height.
 /// It guarantees to be atomic, meaning the view is immutable to outside modifications.
 pub trait AtomicView: Send + Sync {
-    /// The type of the storage view.
-    type View;
+    /// The type of the latest storage view.
+    type LatestView;
 
     /// The type used by the storage to track the commitments at a specific height.
     type Height;
 
     /// Returns the latest block height.
     fn latest_height(&self) -> Option<Self::Height>;
 
-    /// Returns the view of the storage at the given `height`.
-    fn view_at(&self, height: &Self::Height) -> StorageResult<Self::View>;
-
     /// Returns the view of the storage for the latest block height.
-    fn latest_view(&self) -> Self::View;
+    fn latest_view(&self) -> StorageResult<Self::LatestView>;
+}
+
+/// Provides a view of the storage at the given height.
+pub trait HistoricalView: AtomicView {
```

Comment:
Since `HistoricalView` is a subtrait of `AtomicView`, it'd like that the comment here (and possibly on `AtomicView` as well) would mention that `AtomicView` doesn't mean the latest version here.